mediapipe/tasks/ios/test/text/text_embedder/MPPTextEmbedderTests.m

```diff
@@ -0,0 +1,142 @@
+                                                                       error:&error];
+  XCTAssertNil(textEmbedder);
+
+  NSError *expectedError = [NSError
+      errorWithDomain:kExpectedErrorDomain
+                 code:MPPTasksErrorCodeInvalidArgumentError
+             userInfo:@{
+               NSLocalizedDescriptionKey :
+                   @"INVALID_ARGUMENT: ExternalFile must specify at least one of 'file_content', "
+                   @"'file_name', 'file_pointer_meta' or 'file_descriptor_meta'."
+             }];
+  AssertEqualErrors(error, expectedError);
+}
+
+- (void)testEmbedWithBertSucceeds {
+  MPPTextEmbedder *textEmbedder =
+      [self textEmbedderFromModelFileWithName:kBertTextEmbedderModelName];
+
+  MPPEmbedding *embedding1 = [self assertFloatEmbeddingResultsOfEmbedText:kText1
+                                                        usingTextEmbedder:textEmbedder
+                                                                 hasCount:512
+                                                               firstValue:20.057026f];
+
+  MPPEmbedding *embedding2 = [self assertFloatEmbeddingResultsOfEmbedText:kText2
+                                                        usingTextEmbedder:textEmbedder
+                                                                 hasCount:512
+                                                               firstValue:21.254150f];
+  NSNumber *cosineSimilarity = [MPPTextEmbedder cosineSimilarityBetweenEmbedding1:embedding1
+                                                                    andEmbedding2:embedding2
+                                                                            error:nil];
+  XCTAssertEqualWithAccuracy(cosineSimilarity.doubleValue, 0.96386, kDoubleDiffTolerance);
+}
+
+- (void)testEmbedWithRegexSucceeds {
+  MPPTextEmbedder *textEmbedder =
+      [self textEmbedderFromModelFileWithName:kRegexTextEmbedderModelName];
+
+  MPPEmbedding *embedding1 = [self assertFloatEmbeddingResultsOfEmbedText:kText1
+                                                        usingTextEmbedder:textEmbedder
+                                                                 hasCount:16
+                                                               firstValue:0.030935612f];
+
+  MPPEmbedding *embedding2 = [self assertFloatEmbeddingResultsOfEmbedText:kText2
+                                                        usingTextEmbedder:textEmbedder
+                                                                 hasCount:16
+                                                               firstValue:0.0312863f];
+
+  NSNumber *cosineSimilarity = [MPPTextEmbedder cosineSimilarityBetweenEmbedding1:embedding1
+                                                                    andEmbedding2:embedding2
+                                                                            error:nil];
+  XCTAssertEqualWithAccuracy(cosineSimilarity.doubleValue, 0.999937f, kDoubleDiffTolerance);
+}
```

Comment:
Can you also port the newly added test "SucceedsWithMobileBertAndDifferentThemes"?

Reply:
Fixed. Added a test for quantised embedding as well.